.github/workflows/check-pr.yml

```diff
@@ -26,3 +29,148 @@ jobs:
         run: npm ci
       - name: Run validation and tests
         run: npm pack
+
+      - name: Trigger Workflows
+        uses: peter-evans/repository-dispatch@v2
+        with:
+          token: ${{ secrets.WORKFLOW_TRIGGER_SECRET }}  # or use a custom token with proper permissions
+          repository: rdkcentral/firebolt-apis 
+          event-type: trigger-workflow
+          client-payload: '{"OPENRPC_PR_BRANCH": "${{ github.event.pull_request.head.ref }}"}'
+
+      - name: Store the workflow run ID
+        run: |
+          echo "Waiting for Repo2 to finish execution"
+  wait-for-sdk-generation:
+    needs: release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Poll Firebolt-api for JavaScript SDK generation
+        id: poll-javascript-sdk
+        run: |
+          TOKEN="${{ secrets.WORKFLOW_TRIGGER_SECRET }}"
+          REPO_OWNER="rdkcentral"
+          REPO_NAME="firebolt-apis"
+          WORKFLOW_NAME="MFOS standalone sanity report - CORE,MANAGE,DISCOVERY"
+
+          while true; do
```

Comment:
We should add some sort of timeout to this to avoid the job running forever.

Reply:
Added timeout logic